src/main/java/gift/service/MemberService.java

```diff
@@ -0,0 +1,59 @@
+package gift.service;
+
+import gift.dto.MemberDto;
+import gift.util.JwtUtility;
+import gift.util.TokenBlacklist;
+import gift.model.Member;
+import gift.repository.MemberRepository;
+import jakarta.validation.Valid;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.stereotype.Service;
+import java.util.NoSuchElementException;
+
+@Service
+public class MemberService {
+    private final MemberRepository memberRepository;
+    private final TokenBlacklist tokenBlacklist;
+
+    public MemberService(MemberRepository memberRepository, TokenBlacklist tokenBlacklist) {
+        this.memberRepository = memberRepository;
+        this.tokenBlacklist = tokenBlacklist;
+    }
+
+    public String register(@Valid MemberDto memberDto) {
+        memberRepository.findByEmail(memberDto.getEmail())
+                .ifPresent(existingMember -> {
+                    throw new DuplicateKeyException("이미 존재하는 이메일입니다.");
+                });
+        Member member = new Member(memberDto.getEmail(), memberDto.getPassword());
+        Member savedMember = memberRepository.save(member);
+        String token = JwtUtility.generateToken(savedMember.getEmail());
+        Member updatedMember = new Member(savedMember, token);
+        memberRepository.save(updatedMember);
+        return token;
+    }
```

Comment:
```suggestion
    public String register(@Valid MemberDto memberDto) {
        validateNewMember(memberDto);
        String token = JwtUtility.generateToken(memberDto.getEmail());
        Member member = new Member(
                memberDto.getEmail(),
                memberDto.getPassword(),
                token);
        memberRepository.save(member);
        return token;
    }
```

이렇게 로직을 개선해보면 어떨까요?
조금 더 간결해보이지 않나요?
(가장 위에 있는 `validateNewMember` 는 앞서 남긴 메서드 추출 리뷰를 반영한 부분이에요)

뿐만 아니라 위의 로직에서는
`memberRepository.save(member);` 이후
`member.getId()` 를 하면 id 가 제대로 된 값으로 나올거예요. (한 번 해보세요)
어째서 이렇게만 해도 그게 가능할까요?! 🤔

Reply:
👍👍👍참고해서 개선하겠습니다!!
어떻게 가능한지는 열심히 공부하고 나서 다시 코멘트 남기겠습니다.